domain/schema/model/sql/0015-charm.sql

```diff
@@ -307,17 +307,17 @@ CREATE INDEX idx_charm_relation_charm
 ON charm_relation (charm_uuid);
 
 CREATE TABLE charm_extra_binding (
+    uuid TEXT NOT NULL PRIMARY KEY,
     charm_uuid TEXT NOT NULL,
     "key" TEXT NOT NULL,
     name TEXT,
     CONSTRAINT fk_charm_extra_binding_charm
     FOREIGN KEY (charm_uuid)
-    REFERENCES charm (uuid),
-    PRIMARY KEY (charm_uuid, "key", name)
+    REFERENCES charm (uuid)
 );
 
 CREATE INDEX idx_charm_extra_binding_charm
-ON charm_extra_binding (charm_uuid);
```

Comment:
You don't need this index if you're using a uuid.

Reply:
We also have the unique index on `charm_relation`, though key is synonymous with name so just drop name.

Reply:
We've dropped key for charm_resource. Will do that here.